test/gpu_tests.jl

```diff
@@ -0,0 +1,373 @@
+
+import CloudMicrophysics
+import CLIMAParameters
+import Thermodynamics
+
+const CM = CloudMicrophysics
+const CP = CLIMAParameters
+const TD = Thermodynamics
+
+struct EarthParameterSet <: CP.AbstractEarthParameterSet end
+const param_set = EarthParameterSet()
+
+const liquid = CM.Microphysics_1M.LiquidType()
+const ice = CM.Microphysics_1M.IceType()
+const rain = CM.Microphysics_1M.RainType()
+const snow = CM.Microphysics_1M.SnowType()
+
+if get(ARGS, 1, "Array") == "CuArray"
+    using CUDA
+    ArrayType = CUDA.CuArray
+    CUDA.allowscalar(false)
+    device(::Type{T}) where {T <: CuArray} = CUDADevice()
+else
+    ArrayType = Array
+    device(::Type{T}) where {T <: Array} = CPU()
+end
+
+@show ArrayType
+
+@kernel function test_aerosol_activation_kernel!(
+    param_set,
+    output::AbstractArray{FT},
+    r,
+    stdev,
+    N,
+    ϵ,
+    ϕ,
+    M,
+    ν,
+    ρ,
+    κ,
+) where {FT}
+
+    i = @index(Group, Linear)
+
+    # atmospheric conditions (taken from aerosol activation tests)
+    T::FT = 294.0       # air temperature K
+    p::FT = 100000.0    # air pressure Pa
+    w::FT = 0.5         # vertical velocity m/s
+    p_vs::FT = TD.saturation_vapor_pressure(param_set, T, TD.Liquid())
+    q_vs::FT = 1 / (1 - CP.Planet.molmass_ratio(param_set) * (p_vs - p) / p_vs)
+    # water vapor specific humidity (saturated)
+    q = TD.PhasePartition(q_vs, FT(0.0), FT(0.0))
+
+    @inbounds begin
+        mode_B = CM.AerosolModel.Mode_B(
+            FT(r[i]),
+            FT(stdev[i]),
+            FT(N[i]),
+            (FT(1.0),),
+            (FT(ϵ[i]),),
+            (FT(ϕ[i]),),
+            (FT(M[i]),),
+            (FT(ν[i]),),
+            (FT(ρ[i]),),
+            1,
+        )
+        mode_κ = CM.AerosolModel.Mode_κ(
+            FT(r[i]),
+            FT(stdev[i]),
+            FT(N[i]),
+            (FT(1.0),),
+            (FT(1.0),),
+            (FT(M[i]),),
+            (FT(κ[i]),),
+            1,
+        )
+
+        aerosol_distr_B = CM.AerosolModel.AerosolDistribution((mode_B,))
+        aerosol_distr_κ = CM.AerosolModel.AerosolDistribution((mode_κ,))
+
+        output[1, i] = CM.AerosolActivation.mean_hygroscopicity_parameter(
+            param_set,
+            aerosol_distr_B,
+        )[1]
+        output[2, i] = CM.AerosolActivation.mean_hygroscopicity_parameter(
+            param_set,
+            aerosol_distr_κ,
+        )[1]
+
+        output[3, i] = CM.AerosolActivation.total_N_activated(
+            param_set,
+            aerosol_distr_B,
+            T,
+            p,
+            w,
+            q,
+        )
+        output[4, i] = CM.AerosolActivation.total_N_activated(
+            param_set,
+            aerosol_distr_κ,
+            T,
+            p,
+            w,
+            q,
+        )
+
+        output[5, i] = CM.AerosolActivation.total_M_activated(
+            param_set,
+            aerosol_distr_B,
+            T,
+            p,
+            w,
+            q,
+        )
```

Comment:
```suggestion
        args = (T, p, w, q)
        output[3, i] = CM.AerosolActivation.total_N_activated(param_set, aerosol_distr_B, args...)
        output[4, i] = CM.AerosolActivation.total_N_activated(param_set, aerosol_distr_κ, args...)
        output[5, i] = CM.AerosolActivation.total_M_activated(param_set, aerosol_distr_B, args...)
```

Can you reduce number of lines with this? (I haven't tried against the formatter)

Reply:
Oh, and you could apply to `output[6, i]`, too